### DISCUSSION TITLE
Artisan not launching on Windows 11

### DISCUSSION THREAD

**Discussion by dpac151** [artisan-roaster-scope/artisan#1347] [Q&A]
Good afternoon everyone.  I recently downloaded Artisan (version artisan win -2.8.4).  The setup wizard did confirm it was installed successfully.  When I click on the icon on my desktop, I get the loading icon as if it's going to launch, but after about 5-10 seconds it disappears and nothing happens.  I tried right clicking and selecting "open" and I get the same results.  I'm running Windows 11home with a 64-bit operating system, x64-based processor. 

Anyone have any ideas on how to troubleshoot this or able to give me some support?

**Comment by MAKOMO** [comment]
What happens if you download, install and start the current version (v2.10.0)?

**Reply by dpac151** [reply]
Just tried, same result.  I haven't tried the "win-legacy" version, could that make a difference?

**Reply by MAKOMO** [reply]
Do you install Artisan as Admin user? It might be that the installer is not allowed to fully install the software. However, what do I know, I am never used Windows...

**Reply by dpac151** [reply]
Yes it's installed as admin user.  I've also tried the "Run as administrator" function with no luck.  Any idea who I can reach out to for help with Windows?

**Comment by roasterdave** [comment]
Have you tried rebooting Windows?  Artisan does not require that, but sometimes Windows gets itself into a state and needs to reboot to clear things out.  

Are you running a third party antivirus software that could be interfering?  You can try disabling it to see if Artisan will start up.  Note that Windows' built-in virus and threat protection software is happy with Artisan.  

Start Artisan then open Windows Task Manager.  Do you see Artisan listed on the Process tab?

Please report back on these three items so we can narrow down what is happening.

_I am traveling without my Windows 11 machine and in a location where connectivity is spotty.  I may be slow to respond over the next week._

**Reply by dpac151** [reply]
Hi Dave,
1.)Have you tried rebooting Windows? Artisan does not require that, but sometimes Windows gets itself into a state and needs to reboot to clear things out.
**-Yes I've tried rebooting.** 

2.) Are you running a third party antivirus software that could be interfering? You can try disabling it to see if Artisan will start up. Note that Windows' built-in virus and threat protection software is happy with Artisan.
**-Just the Windows defender.  I did get a prompt during the install, I just clicked "run anyway" screen shot attached.** 
![image](https://github.com/artisan-roaster-scope/artisan/assets/153764769/06d7cefb-bf7c-48d3-a07d-cbc3cccdce85)


3.) Start Artisan then open Windows Task Manager. Do you see Artisan listed on the Process tab?
**-This was interesting.  I opened the processes tab, clicked the artisan app.  While the loading icon was spinning it did pop up momentarily on the processes tab, but once the loading icon disappears it drops off the processes list.  Screen shot attached.**
![image](https://github.com/artisan-roaster-scope/artisan/assets/153764769/9eb6930b-b97d-4811-a1f2-2f54226b331e)

![image](https://github.com/artisan-roaster-scope/artisan/assets/153764769/ac8b65a7-eb4c-4398-afe5-c870a43da14a)

**Reply by poundy** [reply]
that screenshot of task manager looks to be from when the artisan installer is running? Can you confirm? 
I would have also expected that the app appears in the "Apps" section of Task Manager not the background processes. 

I suspect what is happening is an app crash of some form. To review that it might be worth looking in your event log. If you want to send that to me privately, can you save out the "System" and "Application" event logs to start, and pop the files on onedrive or similar and send me a link to it? brett underscore pound at hotmail.com is my email address. I'll review and see if there's any interesting signs there that I can see.
![image](https://github.com/artisan-roaster-scope/artisan/assets/279966/5ad219c6-6345-457a-8734-0043da9d1750)

**Reply by poundy** [reply]
can you also confirm, have you ever had Artisan running before, either on this PC or another? And is this PC just a generic "home" computer or is it one that you use with corporate credentials in any way?

**Reply by poundy** [reply]
Just for clarity, the LAST PICTURE that says "Artisan Installer" and "Completing Artisan setup" is the one that seemed out of sequence to me...

**Reply by roasterdave** [reply]
Here is one thing to try.  Download and install the [continuous build](https://github.com/artisan-roaster-scope/artisan/releases/tag/continuous).  Once the install is complete double click the Artisan shortcut and immediately after press the SHIFT and ALT keys.  Hold them down until Artisan starts.  If this is successful you should be able to start Artisan normally from then on.  

If Artisan does not start, the System and Application event logs requested by @poundy will be very useful.

**Reply by dpac151** [reply]
Good morning all,

@roasterdave - uninstalled the previous version and installed the continuous build.  Same result while holding shift and alt.  

@poundy  thanks for the insight.  I cleared the log and attempted launch again twice.  Here is a snip.  Also below is the details of the "information" and "error" that populated.  This is my personal laptop and has never ran artisan before.  I've always been able to download applications without issue. 
![image](https://github.com/artisan-roaster-scope/artisan/assets/153764769/1158b67d-4570-430b-8531-f67aaf8d2602)

**"information"**
Fault bucket 1589880630299592992, type 5
Event Name: BEX64
Response: Not available
Cab Id: 0

Problem signature:
P1: artisan.exe
P2: 2.10.1.0
P3: 657f48c5
P4: ntdll.dll
P5: 10.0.22621.2506
P6: bced4b82
P7: 00000000000a43b0
P8: c0000409
P9: 000000000000000a
P10: 

Attached files:
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER.a5148c16-9949-42c1-877e-b24c44a9c44a.tmp.mdmp
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER.c8070f49-b364-4bf4-b70f-1d28b2bc235f.tmp.WERInternalMetadata.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER.6772436e-87bc-4b78-b101-b948c03355fe.tmp.csv
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER.21993c1c-f7e0-433a-9ca5-e75862ceadd4.tmp.txt
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER.adfe0ea5-91dd-4078-943f-9970c9c1b816.tmp.xml

These files may be available here:
\\?\C:\ProgramData\Microsoft\Windows\WER\ReportArchive\AppCrash_artisan.exe_3dbbbee2e0b02ce7166647e6ea66e1f3956488_361ca4e0_97c979a5-9e30-4b2b-8936-8860b555aed0

Analysis symbol: 
Rechecking for solution: 0
Report Id: 704aeec6-cdb2-40dd-942b-95fd9e214b96
Report Status: 268435456
Hashed bucket: 185998767c3d5da4861064027c480920


**"Error"**
Faulting application name: artisan.exe, version: 2.10.1.0, time stamp: 0x657f48c5
Faulting module name: ntdll.dll, version: 10.0.22621.2506, time stamp: 0xbced4b82
Exception code: 0xc0000409
Fault offset: 0x00000000000a43b0
Faulting process id: 0x0x2558
Faulting application start time: 0x0x1DA31D481E82207
Faulting application path: C:\Program Files\Artisan\artisan.exe
Faulting module path: C:\WINDOWS\SYSTEM32\ntdll.dll
Report Id: 704aeec6-cdb2-40dd-942b-95fd9e214b96
Faulting package full name: 
Faulting package-relative application ID:

**Reply by dpac151** [reply]
![image](https://github.com/artisan-roaster-scope/artisan/assets/153764769/4d1b1600-8504-4805-8a4c-13b432d1e17d)

**Reply by dpac151** [reply]
Good morning @poundy  @roasterdave, just checking in on this one.  Let me know if these screenshots are helpful.  Also willing to jump on a quick call or screen share if that makes sense.  Thanks again for everyone's input so far!

**Reply by roasterdave** [reply]
Have you ever successfully run Artisan on this machine in the past? 

The failure report indicates an `Exception code: 0xc0000409`.  Though error codes can be misleading, generally this one is due to some form of system issue and is thus outside of Artisan itself.  

There is conflicting information on the web about what causes Exception code 0xc0000409 and how to fix it.  Be cautious when doing a web search.  This [article](https://www.minitool.com/news/0xc0000409.html) was written for another application running on Windows 10 and provides a number of steps to take to fix an Exception Code 0xc0000409.  This [article](https://softwaretested.com/windows/getting-error-code-0xc0000409-on-windows/) applies to Windows 11 and also provides a similar series of actions to try to correct it.  Warning these remedies are not all pretty. 

I suggest reading these articles carefully, then at a minimum run the [System File Checker](https://support.microsoft.com/en-us/windows/using-system-file-checker-in-windows-365e0031-36b1-6031-f804-8fd86e0ef4ca) to see if it finds anything.  Next try Artisan after a [clean boot](https://support.microsoft.com/en-us/topic/how-to-perform-a-clean-boot-in-windows-da2f9573-6eec-00ad-2f8a-a97a1807f3dd#ID0EBBD=Windows_11).  Let us know how it goes.

**Reply by poundy** [reply]
yes, my investigation showed that the error message most likely indicates a stack corruption. If this was a real issue in Artisan it would have been reported more than just your single instance (so far).  It is possible that this might well be a new issue that is in Artisan that is causing this corruption when the app has never been installed before, as I understand that this is your situation ("This is my personal laptop and has never ran artisan before") but I can't test nor verify that.

**Comment by MillerOnTap** [comment]
fyi... Just attempted a new install of artisan scope v 2.10.0. Installation went fine, but upon running artisan.exe the executable was immediately put into quarantine. This was with McAfee. McAfee gave a virus hash of "6ed2c64d5a0a22ba4ebf79d942c896e3fd4319018ed63f991f9e52d9e2582e55" and a threat description can be viewed here: https://www.virustotal.com/gui/file/6ed2c64d5a0a22ba4ebf79d942c896e3fd4319018ed63f991f9e52d9e2582e55/detection

Previous Version 2.8.4 installs and runs just fine.

**Reply by roasterdave** [reply]
Thanks for the report.  Virus scanners can be very touchy, especially when widely used open source libraries are used to build code that is malicious.  Other programs using those libraries can be flagged as well.  

Please try the [continuous build](https://github.com/artisan-roaster-scope/artisan/releases/tag/continuous).  As of this writing, [VirusTotal ](https://www.virustotal.com/gui/file/4d9c582b11edb3fb19432dde8846912dcb6b96d44cec9ed6f84021703334272d) does not show McAfee flagging artisan.exe.

In the future, please open a new discussion topic to report a new problem or symptom.  It helps keep the replies and discussion clear and separated.
